src/wp-includes/option.php

```diff
@@ -258,14 +258,24 @@ function get_option( $option, $default_value = false ) {
  *
  * @param array $options An array of option names to be loaded.
  */
-function wp_load_options( $options ) {
+function wp_prime_option_caches( $options ) {
 	$alloptions     = wp_load_alloptions();
-	$cached_options = wp_cache_get_multiple( $options, 'options' );
+	$cache_keys     = (array) $options;
+	$cache_keys[]   = 'notoptions';
+	$cached_options = wp_cache_get_multiple( $cache_keys, 'options' );
+
+	// Store notoptions, remove from cached options.
+	$notoptions = $cached_options['notoptions'];
+	unset( $cached_options['notoptions'] );
 
 	// Filter options that are not in the cache.
 	$options_to_prime = array();
 	foreach ( $options as $option ) {
-		if ( ( ! isset( $cached_options[ $option ] ) || ! $cached_options[ $option ] ) && ! isset( $alloptions[ $option ] ) ) {
+		if (
+			( ! isset( $cached_options[ $option ] ) || ! $cached_options[ $option ] )
```

Comment:
@peterwilsoncc This is not actually part of the bug ticket you raised, but now that I see this again, I am wondering whether there's another problem here. Shouldn't this line rather be:
```suggestion
			( ! isset( $cached_options[ $option ] ) || false === $cached_options[ $option ] )
```

The reason I wonder about that is that normally, only a value of `false` is interpreted as a cache miss. I think the current logic here would for example result in a query even for an option value of `0` which could be a valid value and should be a cache hit as far as I can tell.

If that's the case, maybe we could fix it in this PR too?